### PR TITLE
Added metastatus -vv

### PR DESCRIPTION
**Pull request by mjksmith**
`paasta metastatus -vv` discovers all location attributes for the slaves in the cluster and displays resource utilization (like `-vvv`) for each location grouping that has more than one value in that cluster

### CODE REVIEW

**Comment by solarkennedy** [comment]
Can you paste what this look like?

**Comment by mjksmith** [comment]
Sample output from a pasta slave on nova-devc:

```
Mesos Status: OK
  quorum: masters: 3 configured quorum: 2
  frameworks:
    framework: chronos-2.4.0 count: 1
    framework: marathon count: 1
  CPUs: 4.60 / 9 in use (51.11%)
  Memory: 21.37 / 37.59GB in use (56.85%)
  tasks: running: 15 staging: 0 starting: 0
  slaves: active: 3 inactive: 0
    Habitat       CPU free  RAM free
    useast1bdevc  1.60      4124.00
    useast1adevc  2.80      12483.00
Marathon Status: OK
  marathon apps: 6
  marathon tasks: 15
  marathon deployments: 0
Chronos Status: OK
  Enabled chronos jobs: 1
```

File: paasta_tools/paasta_metastatus.py [hunk 1]
```diff
@@ -63,7 +64,30 @@ def get_mesos_cpu_status(metrics):
     return total, used, available
 
 
+def combine_key_value_pair_iterators(iterator1, iterator2, collision_operator, valid_keys=None):
+    """Takes in two iterators of 2-tuples of the form (key, value) and combines them
+    into a dictionary.
+    :param iterator1: the first iterator to merge
+    :param iterator2: the second iterator to merge
+    :param collision_operator: a function that is applied iterator values to resolve key collisions
+    :param valid_keys" an iterator containing a list of keys to merge, or None to permit all keys
```
Comment (Rob-Johnson) [inline paasta_tools/paasta_metastatus.py]: s/"/:

File: paasta_tools/paasta_metastatus.py [hunk 2]
```diff
@@ -63,7 +64,30 @@ def get_mesos_cpu_status(metrics):
     return total, used, available
 
 
+def combine_key_value_pair_iterators(iterator1, iterator2, collision_operator, valid_keys=None):
```
Comment (solarkennedy) [inline paasta_tools/paasta_metastatus.py]: I'm a bit skeptical about these kin of functions, which kinda hint that we may not be returning manageable datastructures back in the first place. Lets talk about this in person.

**Comment by mjksmith** [comment]
@EvanKrall review please (:

File: paasta_tools/paasta_metastatus.py [hunk 3]
```diff
@@ -63,22 +64,42 @@ def get_mesos_cpu_status(metrics):
     return total, used, available
 
 
+def filter_metastatus_valid_keys(dictionary):
```
Comment (Rob-Johnson) [inline paasta_tools/paasta_metastatus.py]: could you rename this? It's really ``filter_mesos_state_metrics``

**Comment by Rob-Johnson** [comment]
lg2m. ship

**Pull request merged**